features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/TestConsole/TestConsole.jsx

```diff
@@ -419,175 +368,184 @@ class TestConsole extends React.Component {
     render() {
         const { classes } = this.props;
         const {
-            swagger, api, securitySchemeType, selectedEnvironment, productionAccessToken, sandboxAccessToken,
-            labels, environments, scopes, username, password, selectedKeyType, serverError, settings, host, baseUrl,
-            loading,
+            api, swagger, securitySchemeType, selectedEnvironment, environments, settings,
+            showToken, apiSettings,
         } = this.state;
-        if (serverError) {
-            return (
-                <Typography variant='h4' className={classes.titleSub}>
-                    {serverError}
-                </Typography>
-            );
-        }
-        if (!api || !swagger || !settings) {
+        const authorizationHeader = 'Internal-Key';
+        if (!api || !securitySchemeType || !selectedEnvironment || !environments || !swagger || !settings
+            || !apiSettings) {
             return <Progress />;
         }
-        let authorizationHeader = api.authorizationHeader ? api.authorizationHeader : 'Authorization';
-        authorizationHeader = 'testkey';
+        const authHeader = `${authorizationHeader}`;
         if (!swagger.openapi) {
-            swagger.host = host;
-            swagger.basePath = baseUrl;
+            for (let i = 0; i < apiSettings.environment.length; i++) {
+                if (apiSettings.environment[i].name === selectedEnvironment) {
+                    const val = apiSettings.environment[i].endpoints.https.split('//')[1];
+                    swagger.host = val;
```

Comment:
Is it always valid to extract the host and construct the request URL ? What if someone has a context?

`https://apis.gate.wso2.com/us/`